File: Tiny_EGT.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

func main() {
    rand.Seed(time.Now().UnixNano())
    fmt.Println("=== Tiny EGT Module Simulation ===")

    for i := 0; i < 10; i++ {
        temperature := 900 + rand.Intn(101) // 900-1000
        fuelTrim := 0
        if temperature > 950 {
            fuelTrim = -10
        }

        line := fmt.Sprintf("Temperature: %d | Fuel Trim: %d", temperature, fuelTrim)
        if temperature > 950 {
            line += " | WARNING: Overtemp!"
        }
        fmt.Println(line)
    }

    fmt.Println("=== Simulation Complete ===")
}

File: Tiny_EGT.py
import random

print("=== Tiny EGT Module Simulation ===")

for _ in range(10):
    temperature = random.randint(900, 1000)
    fuel_trim = -10 if temperature > 950 else 0
    line = f"Temperature: {temperature} | Fuel Trim: {fuel_trim}"
    if temperature > 950:
        line += " | WARNING: Overtemp!"
    print(line)

print("=== Simulation Complete ===")